src/spikeinterface/core/basesorting.py

```diff
@@ -109,18 +109,20 @@ def get_unit_spike_train(
     ):
         segment_index = self._check_segment_index(segment_index)
         segment = self._sorting_segments[segment_index]
-        spike_frames = segment.get_unit_spike_train(
-            unit_id=unit_id, start_frame=start_frame, end_frame=end_frame
-        ).astype("int64")
         if return_times:
-            if self.has_recording():
-                times = self.get_times(segment_index=segment_index)
-                return times[spike_frames]
-            else:
-                t_start = segment._t_start if segment._t_start is not None else 0
-                spike_times = spike_frames / self.get_sampling_frequency()
-                return t_start + spike_times
+            spike_times = segment.get_unit_spike_times(
```

Comment:
Make a `get_unit_spike_times(t_start, t_end, ...)` in `BaseSorting`, otherwise the conversion between frames and times could eb ambicuous (e.g. if there are gaps)